cli/src/template_builder.rs

```diff
@@ -246,15 +246,30 @@ pub type TemplateBuildMethodFnMap<'a, L, T> =
 
 /// Symbol table of methods available in the core template.
 pub struct CoreTemplateBuildFnTable<'a, L: TemplateLanguage<'a>> {
-    string_methods: TemplateBuildMethodFnMap<'a, L, String>,
-    boolean_methods: TemplateBuildMethodFnMap<'a, L, bool>,
-    integer_methods: TemplateBuildMethodFnMap<'a, L, i64>,
-    signature_methods: TemplateBuildMethodFnMap<'a, L, Signature>,
-    timestamp_methods: TemplateBuildMethodFnMap<'a, L, Timestamp>,
-    timestamp_range_methods: TemplateBuildMethodFnMap<'a, L, TimestampRange>,
+    pub string_methods: TemplateBuildMethodFnMap<'a, L, String>,
+    pub boolean_methods: TemplateBuildMethodFnMap<'a, L, bool>,
+    pub integer_methods: TemplateBuildMethodFnMap<'a, L, i64>,
+    pub signature_methods: TemplateBuildMethodFnMap<'a, L, Signature>,
+    pub timestamp_methods: TemplateBuildMethodFnMap<'a, L, Timestamp>,
+    pub timestamp_range_methods: TemplateBuildMethodFnMap<'a, L, TimestampRange>,
     // TODO: add global functions table?
 }
 
+pub fn merge_fn_map<'a, L: TemplateLanguage<'a>, T>(
+    base: &mut TemplateBuildMethodFnMap<'a, L, T>,
+    extension: TemplateBuildMethodFnMap<'a, L, T>,
+) {
+    for (name, function) in extension {
+        if let Some(orig_fn) = base.get_mut(name) {
+            // If one or more extensions provide conflicting function definitions, error out
+            // on invoking said function.
+            *orig_fn = |_, _, _, call| Err(TemplateParseError::ambiguous_function(call));
```

Comment:
Better to `panic()` or take the last one?

The user can't fix the problem, so it's pretty much an implementation bug.

Another option is to pass the original map to extension callback, and let the extension author decide what to do.

Reply:
panic() seems good - I was thinking it might be better to let the user avoid that function, but you're right that it's mostly unactionable by a non-dev